### ISSUE TITLE
Replace print / warnings with proper logging

### ISSUE THREAD

**Issue by jlaura** [USGS-Astrogeology/autocnet#602]
The library is littered with print and warning statements. We do not have proper logging being used anywhere. This is resulting in quite messy output of all types that is hard to do anything with after job runs. 

Proposal:

- Adopt the logging library
- Output all logs as JSON so that they can be used in another tool that ingests / parses / visualizes JSON formatted logs (e.g., loggly or elastic search / Kibana).

**Comment by rbeyer** [comment]
Yes, this project should adopt the Python Standard Library *logging* module, it contains a depth of functionality that would be ludicrous to attempt to home-brew.

I'm neutral on the JSON thing, and consider that an option.  The *logging* module provides a variety of output choices, and if you really wanted them written out in JSON, you can do that via a custom handler mechanism from within the *logging* module.

**Comment by jlaura** [comment]
Finally getting to get some work on this issue! I would like to have the logging setup where each module logs to a module specific logger a la:

```python
import logging
log = logging.getLogger(__name__) # name here is the filename of the module
log.debug('My debug message')
log.error('Something broke....stack trace here.')
```

Then we need to determine how best to configure the application level logger. For the candidate graph, this should happen inside of the CandidateGraph object since that is the primary entry point, e.g., 'the application'. For the network candidate graph this gets a little bit gnarlier because the NCG is the object that orchestrates the jobs, but the acn_submit (and to be pushed acn_submit_single) bin scripts are the app that needs to configure logging. Ultimately, as a user, if I have log messages that should propagate up to the candidate graph objects as I use them, I want to see them. If I have log messages from the submit scripts, I want to write those to stdout so that they are stored on disk post cluster runs.

On the point of cluster runs - it is important to flush to the file (or stdout) after each line so that a crashed cluster job writes logs that can be inspected.

I *think*, we can then do something as simple as `logging.basicConfig(level=os.environ.get("AUTOCNET_LOGLEVEL", "INFO"))` in the `__init__` section of the candidate graphs and as the first item in the bin scripts and we should have the logging that we want to have. One nice snippet that I found for making multiline error logs into single line (for later analysis):

```python
import logging
import os
 
class OneLineExceptionFormatter(logging.Formatter):
    def formatException(self, exc_info):
        result = super().formatException(exc_info)
        return repr(result)
 
    def format(self, record):
        result = super().format(record)
        if record.exc_text:
            result = result.replace("\n", "")
        return result
 
handler = logging.StreamHandler()
formatter = OneLineExceptionFormatter(logging.BASIC_FORMAT)
handler.setFormatter(formatter)
root = logging.getLogger()
root.setLevel(os.environ.get("LOGLEVEL", "INFO"))
root.addHandler(handler)
```

**Comment by gsn9** [comment]
This issue will be solved once we replace all of the warnings in the modules `camera, cg, control, examples, graph, io, matcher, spatial, transformation, utils, and vis` with log warnings. After updating these modules to use logs rather than warnings their corresponding test's must be updated to ensure that the correct information is being logged. 

**---Modules---**
- [ ] camera
- [ ] cg
- [ ] control
- [ ] examples
- [ ] graph
- [ ] io
- [ ] matcher
- [ ] spatial
- [ ] transformation
- [ ] utils
- [ ] vis
